newwebsite/generate.jl

```diff
@@ -0,0 +1,605 @@
+	paths...
+], "/")
+
+# ╔═╡ 35b80456-039e-45bc-963e-5466a3e9c3a7
+@use_task([]) do 
+	sleep(1)
+	error("asdf")
+end
+
+# ╔═╡ a965eb6b-8c70-4986-a7b1-99c820c45716
+@use_task([output_dir, file_server_port]) do
+
+	run(`$(Deno_jll.deno()) run --allow-net --allow-read https://deno.land/std@0.115.0/http/file_server.ts $(output_dir) --cors --port $(file_server_port)`)
+
+end
+
+# ╔═╡ a24bf899-87b0-4a2e-a6d4-30ac2aad4820
+md"""
+# Static assets
+"""
+
+# ╔═╡ 2bba13d3-0c1d-4d17-bd70-526ce70407fb
+md"""
+# Sidebar
+"""
+
+# ╔═╡ 444502c9-33b5-4bb2-9a8d-a8d8e1adb632
+function sidebar_code(book_model)
+    @htl("""
+    <div class="sidebar">
+    <div class="container sidebar-sticky">
+    <div class="sidebar-about">
+    <br>
+    <img src="/assets/MIT_logo.svg" style="width: 80px; height: auto; display: inline">
+    <img src="/assets/julia-logo.svg" style="margin-left:1em; width: 80px; height: auto; display: inline">
+    <div style="font-weight: bold; margin-bottom: 0.5em"><a href="/semesters/">Spring 2021</a> <span style="opacity: 0.6;">| MIT 18.S191/6.S083/22.S092</span></div>
+    <h1><a href="/">Introduction to Computational Thinking</a></h1>
+    <h2>Math from computation, math with computation</h2>
+    <div style="line-height:18px; font-size: 15px; opacity: 0.85">by <a href="http://math.mit.edu/~edelman">Alan Edelman</a>, <a href="http://sistemas.fciencias.unam.mx/~dsanders/">David P. Sanders</a> &amp; <a href="https://people.csail.mit.edu/cel/">Charles E. Leiserson</a></div>
+    </div>
+    <br>
+    <style>
+    </style>
+    <nav class="sidebar-nav" style="opacity: 0.9">
+    <a class="sidebar-nav-item {{ispage /index.html}}active{{end}}" href="/"><b>Welcome</b></a>
+    <a class="sidebar-nav-item {{ispage /reviews/}}active{{end}}" href="/reviews/">Class Reviews</a>
+    <a class="sidebar-nav-item {{ispage /logistics/}}active{{end}}" href="/logistics/">Class Logistics</a>
+    <a class="sidebar-nav-item {{ispage /homework/}}active{{end}}" href="/homework/">Homework</a>
+    <a class="sidebar-nav-item {{ispage /syllabus/}}active{{end}}" href="/syllabus/">Syllabus and videos</a>
+    <a class="sidebar-nav-item {{ispage /installation/}}active{{end}}" href="/installation/">Software installation</a>
+    <a class="sidebar-nav-item {{ispage /cheatsheets/}}active{{end}}" href="/cheatsheets/">Cheatsheets</a>
+    <a class="sidebar-nav-item {{ispage /semesters/}}active{{end}}" href="/semesters/">Previous semesters</a>
+    <br>
+    $(map(enumerate(book_model)) do (chapter_number, chap)
+		@htl("""
+		<div class="course-section">Module $(chapter_number): $(chap.title)</div>
+		
+		$(map(enumerate(chap.contents)) do (section_number, section)
+
+			notebook_name = 
+				basename(without_pluto_file_extension(section.notebook_path))
+		    @htl("""
+		    <a class="sidebar-nav-item {{ispage /$notebook_name/}}active{{end}}" href="/$notebook_name/"><b>$(chapter_number).$(section_number)</b> - <em>$(section.name)</em></a>
+		    """)
```

Comment:
```suggestion
			notebook_name = 
				basename(without_pluto_file_extension(section.notebook_path))
			notebook_id = flatten_path(without_pluto_file_extension(section.notebook_path))
		    @htl("""
		    <a class="sidebar-nav-item {{ispage /$notebook_name/}}active{{end}}" href="/$notebook_id/"><b>$(chapter_number).$(section_number)</b> - <em>$(section.name)</em></a>
		    """)
```

Reply:
some explanation, since there doesn't seem to be highlighting of changed words:

This fixes links to notebooks in the sidebar.

It uses `notebook_id` instead of `notebook_name` for the href. `notebook_id` is the name you assign to the build of the notebooks, e.g. `notebooks_week3_newton_method`.